src/java/org/apache/cassandra/index/sasi/plan/SASIIndexQueryPlan.java

```diff
@@ -0,0 +1,51 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.cassandra.index.sasi.plan;
+
+import javax.annotation.Nullable;
+
+import org.apache.cassandra.db.filter.RowFilter;
+import org.apache.cassandra.index.SingletonIndexQueryPlan;
+import org.apache.cassandra.index.sasi.SASIIndex;
+
+public class SASIIndexQueryPlan extends SingletonIndexQueryPlan
+{
+    private SASIIndexQueryPlan(SASIIndex index, RowFilter postIndexFilter)
+    {
+        super(index, postIndexFilter);
+    }
+
+    @Nullable
+    public static SASIIndexQueryPlan create(SASIIndex index, RowFilter rowFilter)
+    {
+        for (RowFilter.Expression e : rowFilter.getExpressions())
+        {
+            if (index.supportsExpression(e.column(), e.operator()))
+                return new SASIIndexQueryPlan(index, index.getPostIndexQueryFilter(rowFilter));
+        }
+
+        return null;
+    }
+
+    @Override
+    public boolean supportsReplicaFilteringProtection(RowFilter rowFilter)
+    {
+        return false;
```

Comment:
In SASI's case, this is `false` because it supports tokenization, etc. correct? (Having to jog the ol' memory...)

Reply:
iirc, we never intended to support SASI..